src/handlers/notify_zulip.rs

```diff
@@ -172,3 +173,66 @@ pub(super) async fn handle_input<'a>(
 
     Ok(())
 }
+
+fn replace_team_to_be_nominated(labels: &Vec<Label>, msg: String) -> String {
+    let teams = labels
+        .iter()
+        .map(|label| &label.name)
+        .filter_map(|label| label.strip_prefix("T-"))
+        .collect::<Vec<&str>>();
+
+    // - If a single team label is found, replace the placeholder with that one
+    // - If multiple team labels are found and one of them is "compiler", pick that one
+    //   (currently the only team handling these Zulip notification)
+    // - else, do nothing
+    if teams.len() == 1 {
+        msg.replace("{team}", teams[0])
```

Comment:
```suggestion
    if let [team] = &*teams {
        msg.replace("{team}", team)
```

Reply:
nit: I find the suggested syntax a bit obscure. What is the difference between the two versions? IIUC avoiding a `.len()` saves a little memory? Like one an allocation less? (trying to learn something :slightly_smiling_face: )

Reply:
I see what you mean (esp. the `&*` which I sometimes replace with `.as_slice()` for readability).

It's not about allocations but panicking. By using a slice pattern, one can quickly see that this code cannot panic (unless `-Zoom=panic` ofc ^^) but with the check and the explicit indexing, it's “less obvious”. In this case, admittedly it doesn't make a huge difference but it's a nice pattern to follow. It allows for [local reasoning](https://stackoverflow.com/questions/61207079/what-is-local-reasoning-exactly). You wouldn't write `if x.is_some() { x.unwrap() }` either for example.

Reply:
awesome, TIL! Thanks for the additional context